### PR TITLE
#158561694 Create profile page 

### PR DESCRIPTION
**Pull request by chadwalt**
#### What does this PR do?
Adds a user profile page
#### Description of the task to be completed?
This PR adds a profile page where a user can update his/her information. 
#### How should this be manually tested?
This can be tested by navigating to the profile page `http://127.0.0.1:9393/profile`
#### What are the relevant pivotal tracker stories?
[158561694](https://www.pivotaltracker.com/story/show/158561694)
#### Screenshots
screen shot 2018-07-15 at 23 02 47
screen shot 2018-07-15 at 23 00 43

### CODE REVIEW

File: app/controllers/signup_controller.rb [hunk 1]
```diff
@@ -21,7 +21,8 @@ class SignupController < ApplicationController
 
     if @user.save
       session.delete(:error) if session[:error]
-      redirect to('/dashboard')
+      session[:email] = @user.email
```
Comment (igbanam) [inline app/controllers/signup_controller.rb]: Dunno what the goal is here, but it's generally bad practice to pass user information like this in plain sight

File: app/controllers/profile_controller.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    redirect '/profile'
+  end
+
+  post '/change_password' do
+    user = User.find_by(email: session[:email])
+
+    if user.authenticate(params[:user]['old_password'])
+      if params[:user]['password'] == params[:user]['password_confirmation']
+        user.update_attributes(password_digest: params[:user]['old_password'])
```
Comment (igbanam) [inline app/controllers/profile_controller.rb]: this is room for a potential security breach. is there any way you can sanitize these inputs before passing them into your database?

File: app/controllers/profile_controller.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    User.where(email: session[:email]).update(params[:user])
+    session[:user] = params[:user]
+    session[:success] = 'Successfully Saved'
+
+    redirect '/profile'
+  end
+
+  post '/change_password' do
+    user = User.find_by(email: session[:email])
```
Comment (igbanam) [inline app/controllers/profile_controller.rb]: prone to SQL injection

if someone enters `'; drop users;'` as an input, your application would be in a rather interesting state.

My advise: sanitize all inputs

File: app/controllers/profile_controller.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+      File.open("./public/images/#{file_name}", 'wb') do |f|
+        f.write(temp_file.read)
+      end
+
+      params[:user]['image'] = file_name
+    end
+
+    User.where(email: session[:email]).update(params[:user])
+    session[:user] = params[:user]
```
Comment (igbanam) [inline app/controllers/profile_controller.rb]: there should be a better way to do this. are you trying to get data into the view?

Comment (chadwalt) [reply app/controllers/profile_controller.rb]: I have resorted to using variables.

File: app/controllers/signup_controller.rb (hunk 1 above)
Comment (chadwalt) [reply app/controllers/signup_controller.rb]: I have addressed this

File: app/controllers/profile_controller.rb (hunk 3 above)
Comment (chadwalt) [reply app/controllers/profile_controller.rb]: I have created a helper to escape_html

File: app/controllers/profile_controller.rb (hunk 2 above)
Comment (chadwalt) [reply app/controllers/profile_controller.rb]: I have addressed this

**Pull request merged**